pkg/gohai/filesystem/filesystem_nix_test.go

```diff
@@ -3,145 +3,356 @@
-		dfOptions = oldOptions
-	})
-}
 
-func TestSlowDf(t *testing.T) {
-	withDfCommand(t, "sleep", "5")
-	dfTimeout = 20 * time.Millisecond // test faster
-	defer func() { dfTimeout = 2 * time.Second }()
+	caser := cases.Title(language.English)
+	for _, tc := range testCases {
+		tc := tc
+		t.Run("TestIgnoringOfFSType/"+caser.String(tc.FSType), func(t *testing.T) {
+			inputMounts := []*mountinfo.Info{
+				{
+					Source:     tc.FSName,
+					FSType:     tc.FSType,
+					Mountpoint: randString(),
+				},
+				newTestInputMountinfo(randString()),
+				newTestInputMountinfo(randString()),
+				{
+					Source:     tc.FSName,
+					FSType:     tc.FSType,
+					Mountpoint: randString(),
+				},
+				newTestInputMountinfo(randString()),
+			}
 
-	_, err := getFileSystemInfo()
-	require.ErrorContains(t, err, "df failed to collect filesystem data")
+			expectedMounts := make([]MountInfo, 0, len(inputMounts))
+			for _, mount := range inputMounts {
+				// We only care about excluding specific FS types that we are doing the
+				// test for and that are marked as excluded
+				if mount.FSType == tc.FSType {
+					if !tc.Included {
+						continue
+					}
+				}
+
+				expectedMount := MountInfo{
+					Name:      mount.Source,
+					MountedOn: mount.Mountpoint,
+					SizeKB:    1,
+				}
+
+				expectedMounts = append(expectedMounts, expectedMount)
+			}
+
+			mounts, err := getFileSystemInfoWithMounts(inputMounts, mockFSInfo)
+			require.NoError(t, err)
+
+			require.Equal(t, len(expectedMounts), len(mounts))
+			assert.ElementsMatch(t, mounts, expectedMounts)
+		})
+	}
 }
 
-func TestOldMacosDf(t *testing.T) {
-	// from https://apple.stackexchange.com/questions/263437/df-hide-ifree-iused-512-blocks-customize-column-format-dont-show-inode-info
-	withDfCommand(t, "sh", "-c", `
-		echo 'Filesystem                                 1K-blocks       Used Available Capacity  iused    ifree %iused  Mounted on';
-		echo '/dev/disk0s2                               975093952  719904648 254677304    74% 90052079 31834663   74%   /';
-		echo 'devfs                                            368        368         0   100%      637        0  100%   /dev';
-		echo 'map -hosts                                         0          0         0   100%        0        0  100%   /net';
-		echo 'map -static                                        0          0         0   100%        0        0  100%   /Volumes/Large';
-	`)
-
-	out, err := getFileSystemInfo()
-	require.NoError(t, err)
-	require.Equal(t, []interface{}{
-		map[string]string{"kb_size": "975093952", "mounted_on": "/", "name": "/dev/disk0s2"},
-		map[string]string{"kb_size": "368", "mounted_on": "/dev", "name": "devfs"},
-		map[string]string{"kb_size": "0", "mounted_on": "/net", "name": "map -hosts"},
-		map[string]string{"kb_size": "0", "mounted_on": "/Volumes/Large", "name": "map -static"},
-	}, out)
+func TestNixMissingMountValues(t *testing.T) {
+	mockFSInfo := newMockFSInfo()
+
+	var testCases = []struct {
```

Comment:
Same here:

```suggestion
	testCases := []struct{
```

Reply:
Done in a028ed4b20e60f57eb5d24d8888cf4312f4ef7c2